internal/statestore/redis.go

```diff
@@ -335,8 +345,15 @@ func (rb *redisBackend) DeleteTicket(ctx context.Context, id string) error {
 
 // IndexTicket indexes the Ticket id for the configured index fields.
 func (rb *redisBackend) IndexTicket(ctx context.Context, ticket *pb.Ticket) error {
+	if ticket == nil {
```

Comment:
We already have validation in the calling code, eg https://github.com/googleforgames/open-match/blob/master/internal/app/frontend/frontend_service.go

Here and elsewhere in your changes we don't need to be validating at this level.  If there's any missing validation, it should be added in the grpc service.